vignettes/higher_order.Rmd

```diff
@@ -0,0 +1,205 @@
+---
+title: "Higher Order Functions"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Higher Order Functions}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+# Introduction
+
+This vignette explains some of the more advanced options of `{admiral}` related
+to higher order functions. A higher order function is a function that takes another
+function as input. By introducing these higher order functions, we intend to give
+the user greater power over our derivation functions, whilst trying to negate the
+need for adding additional `{admiral}` functions or arguments, or the user needing
+many separate steps.
+
+The functions covered here are:
+* `call_derivation()`: Call a single derivation multiple times with some
+parameters/arguments being fixed across iterations and others varying
+* `restrict_derivation()`: Execute a derivation on a subset of the input dataset
+* `slice_derivation()`: The input dataset is split into slices (subsets) and for
+each slice the derivation is called separately. Some or all arguments of the
+derivation may vary depending on the slice.
```

Comment:
The list does not render correctly in html.

Reply:
You'll have to insert a line break before the first list item in order to render properly.

Reply:
Great descriptions Ross. For the slice function, I assume the separately called derivation is the same with different arguments. Or could it also be different derivations per slice?